009-s3-storage-support.md

```diff
@@ -0,0 +1,207 @@
+2. create url using bucket and region (<bucket>.s3.<region>.amazonaws.com)
+3. Create and execute the request in the following format
+
+```http
+DELETE /my-second-image.jpg HTTP/1.1
+Host: <bucket>.s3.<Region>.amazonaws.com
+Date: Wed, 12 Oct 2009 17:50:00 GMT
+Authorization: authorization string
+Content-Type: text/plain        
+```
+https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html#API_DeleteObject_RequestSyntax
+
+### Environment variables
+- AWS_ACCESS_KEY: AWS access key ID
+- AWS_SECRET: AWS secret access key
+- AWS_REGION: Region
+- AWS_BUCKET: Bucket
+
+<!--
+This is the technical portion of the RFC. Explain the design in sufficient detail keeping in mind the following:
+
+- Its interaction with other parts of the system is clear
+- It is reasonably clear how the contribution would be implemented
+- Dependencies on libraries, tools, projects or work that isn't yet complete
+- New API routes that need to be created or modifications to the existing routes (if needed)
+- Any breaking changes and ways in which we can ensure backward compatibility.
+- Use Cases
+- Goals
+- Deliverables
+- Changes to documentation
+- Ways to scale the solution
+
+Ensure that you include examples, code-snippets etc. to allow the community to understand the proposed solution. **It would be best if the examples use naming conventions that you intend to use during the actual implementation so that changes can be suggested early on during the development.**
+
+Write your answer below.
+
+-->
+
+### Prior art
+
+[prior-art]: #prior-art
+
+<!--
+
+Discuss prior art, both the good and the bad, in relation to this proposal. A
+few examples of what this can include are:
+
+- Does this functionality exist in other software and what experience has their
+  community had?
+- For other teams: What lessons can we learn from what other communities have
+  done here?
+- Papers: Are there any published papers or great posts that discuss this? If
+  you have some relevant papers to refer to, this can serve as a more detailed
+  theoretical background.
+
+This section is intended to encourage you as an author to think about the
+lessons from other software, provide readers of your RFC with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us
+whether they are brand new or if it is an adaptation from other software.
+
+Write your answer below.
+-->
+
+### Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+<!-- What parts of the design do you expect to resolve through the RFC process before this gets merged? -->
+
+<!-- Write your answer below. -->
+1. Do we support multiple buckets? If yes, how?
+2. There are options like dual-stack, file versioning, which is yet to be explored, and supporting all those features might be difficult without the SDK, will be much like implementing the whole S3 part of AWS SDK.
```

Comment:
I would start handling file versioning now, anyway appwrite support only write and delete and no update. But it would be nice to still mention it as a future possibility.